### ISSUE TITLE
Migrate controllers and factories to TypeScript and extract services

### ISSUE THREAD

**Issue by liangyuanruo** [opengovsg/FormSG#144]
Blocked by #22 

TODOs
- Migrate controllers/factories to TypeScript
- For each controller, extract relevant business logic into services
- Refactor implementations to use up-to-date patterns where possible e.g. from callbacks to async/await
- Add new Jest tests, or migrate existing tests from Jasmine to Jest

Controllers
- [x] admin-console.server.controller.js (@karrui)
- [ ] admin-forms.server.controller.js (@karrui)
- [x] authentication.server.controller.js (@liangyuanruo)
- [x] core.server.controller.js (@karrui)
- [x] email-submissions.server.controller.js (@mantariksh)
- [ ] encrypt-submissions.server.controller.js
- [x] errors.server.controller.js (@tshuli)
- [x] myinfo.server.controller.js (@mantariksh )
- [x] public-forms.server.controller.js (@karrui, 2/6 done. Blocked by spcp server controller refactor)
- [x] spcp.server.controller.js (@mantariksh)
- [x] submissions.server.controller.js (@mantariksh)
- [x] verification.server.controller.js (@mantariksh)
- [x] webhooks.server.controller.js (@mantariksh)
- [x] user.server.controller

Factories
- [x] aggregate-stats.factory.js (@karrui)
- [x] captcha.factory.js (@karrui)
- [ ] google-analytics.factory.js (@tshuli)
- [x] sms.factory.js (@karrui)
- [x] verified-fields.factory.js (@mantariksh)
- [ ] webhook-verified-content.factory.js

**Comment by liangyuanruo** [comment]
Superseded by #1434, #1435, #1437, #1438.
